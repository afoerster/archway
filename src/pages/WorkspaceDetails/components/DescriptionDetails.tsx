import * as React from 'react';
import { Card } from 'antd';
import Label from './Label';

interface Props {
  description: string;
}

const DescriptionDetails = ({ description }: Props) => (
  <Card
    style={{ display: 'flex', flex: 1 }}
    bodyStyle={{ display: 'flex', flexDirection: 'column', flex: 1 }}>
    <Label>description</Label>
    <div
      style={{
        textAlign: 'center',
        display: 'flex',
        flexDirection: 'column',
        flex: 1,
        alignItems: 'center',
        justifyContent: 'center',
      }}>
      <p style={{ textAlign: 'center', marginBottom: 0 }}>
        {description}
      </p>
    </div>
  </Card>
);

export default DescriptionDetails;
