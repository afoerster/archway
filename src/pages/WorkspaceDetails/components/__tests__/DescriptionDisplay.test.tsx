import * as React from 'react';
import { shallow } from 'enzyme';

import DescriptionDetails from '../DescriptionDetails';

describe('DescriptionDetails', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<DescriptionDetails description="" />);
    expect(wrapper).toMatchSnapshot();
  });
});
