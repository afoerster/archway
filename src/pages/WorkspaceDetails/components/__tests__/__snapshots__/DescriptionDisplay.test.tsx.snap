// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DescriptionDetails renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DescriptionDetails
    description=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "bodyStyle": Object {
        "display": "flex",
        "flex": 1,
        "flexDirection": "column",
      },
      "children": Array [
        <Label>
          description
        </Label>,
        <div
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "center",
              "textAlign": "center",
            }
          }
        >
          <p
            style={
              Object {
                "marginBottom": 0,
                "textAlign": "center",
              }
            }
          >
            
          </p>
        </div>,
      ],
      "style": Object {
        "display": "flex",
        "flex": 1,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "description",
        },
        "ref": null,
        "rendered": "description",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <p
            style={
              Object {
                "marginBottom": 0,
                "textAlign": "center",
              }
            }
          >
            
          </p>,
          "style": Object {
            "alignItems": "center",
            "display": "flex",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
            "textAlign": "center",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "style": Object {
              "marginBottom": 0,
              "textAlign": "center",
            },
          },
          "ref": null,
          "rendered": "",
          "type": "p",
        },
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bodyStyle": Object {
          "display": "flex",
          "flex": 1,
          "flexDirection": "column",
        },
        "children": Array [
          <Label>
            description
          </Label>,
          <div
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "center",
                "textAlign": "center",
              }
            }
          >
            <p
              style={
                Object {
                  "marginBottom": 0,
                  "textAlign": "center",
                }
              }
            >
              
            </p>
          </div>,
        ],
        "style": Object {
          "display": "flex",
          "flex": 1,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "description",
          },
          "ref": null,
          "rendered": "description",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <p
              style={
                Object {
                  "marginBottom": 0,
                  "textAlign": "center",
                }
              }
            >
              
            </p>,
            "style": Object {
              "alignItems": "center",
              "display": "flex",
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "center",
              "textAlign": "center",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
              "style": Object {
                "marginBottom": 0,
                "textAlign": "center",
              },
            },
            "ref": null,
            "rendered": "",
            "type": "p",
          },
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
